### ISSUE TITLE
Falcon sense using just half of the threads after newest update (before was using all threads)

### ISSUE THREAD

**Issue by rafalwoycicki** [marbl/canu#55]
Hi All,

I have updated canu installation yestarday. The strange thing is, that now falcon sense is running just 3 jobs concurrently instead of 24 as before upate. (I have 24 threads machine with 64 gb ram). Each of the jobs is using just 4 threads, so half of the max cores/job. Earlier there were 24 single threaded jobs. I am using the same similar (even less data) but this step 2-correction takes already twice as long and is just in the middle (36/67 jobs) - already 10 hours, before update it took 5 hours for 62 jobs. Do you know what can be a problem?

![screenshot from 2016-02-27 12 56 35](https://cloud.githubusercontent.com/assets/17161885/13372740/9f11999a-dd51-11e5-9d82-52851ee56ac1.png)

[sub_4_50_l52_85_40.report.txt](https://github.com/marbl/canu/files/149321/sub_4_50_l52_85_40.report.txt)

**Comment by skoren** [comment]
The code had quite a few changes including being parallelized by alignment than by read (as the old one was). This means if you have a lot of low coverage reads it may not be able to use all available threads since there aren't enough alignments. I'm guessing you're running on a single machine? On most systems, especially where IO is the bottleneck (like grid systems), the new parallelization provides better CPU utilization. That said, Canu should be using both read-level and alignment-level parallelization if it has more than one falcon_sense partition.

The default CPU usage for falcon_sense shouldn't be 24 threads but 2-8 (depending on your genome size), are you manually adjusting it? If so, I'd suggest leaving the threads unset to use the default which will let you run multiple falcon_sense jobs each using 2-8 cores but on multiple partitions of reads. That should give you full CPU utilization of your machine.

**Comment by rafalwoycicki** [comment]
Hi Sergey,

Thanks a lot for your reply.
I have an 90Mbp genome which went succesfully through 3 WGDs. I took 100x of PacBio P6C4 reads from 4kb to 50 kb of QV>=85
The distribution of reads shows that I have the most of them starting from QV=82 (I was checking this assembly version of correction/assembly also but till now using the PBcR from wgs8.3rc2, but to get similarly high number of reads after correction (comparing to QV>=85 and 40x corrected when haviong 100x_which map to the reference with 99% similarity on >=90% of the read length I had to set the PbCR to correct 60x of reads (total cov 160x) - other way I am getting 1/3 less corrected reads properly aligning to the reference.
Do you think that this restriction to the QV>=85, could have laft me with the reads not aligning too much?
I will try then QV>=83 at 60x.
I attach below the distribution of raw reads. You can see from ti that the library length was in average 9Kbp. So as I play with the number of reads I try to set the lower length of reads to be corrected to 10-11Kbp, so a little above, cause as I have seen (using PBcR) when I go a bit higher so to 13Kbp I got a lot less sequences longer then 10Kbp aligning to reference with >=99% similarity.

I have other question rearding the MHAP 2.0 witch is now as the default. At high sensitivity it is using now 512 hashes, earlier was 756. Also..it is possible somehow to set the details of mhap manually, for example to higher number of hashes?

![all_subreads_distribution](https://cloud.githubusercontent.com/assets/17161885/13380319/366fee9c-de40-11e5-9508-ae29752209d9.png)

**Comment by skoren** [comment]
Canu (and PBcR) try to correct the longest 40X but due to errors in the sequences and minimum coverage requirements, it usually ends up with 25-35X corrected rather than 40X but that coverage should be sufficient for a good assembly. If you want to get exactly 40X corrected, you would need to set the corOutCoverage to a higher value (like you're doing with 60X) but that isn't likely to change your assembly.

The high sensitivity should still be using 768, that was a bug which I fixed. To manually adjust MHAP you'd have to edit the Canu parameters in src/pipelines/canu/OverlapMhap.pm but we've found that the default setting works very well (in fact it can be dropped to 256 hashes without impacting the assembly) so you shouldn't need to tune this.

**Comment by rafalwoycicki** [comment]
Hi Sergey,

Yes, thats what I am changing (corOutCoverage). I am testing different versions of corrections/assembly.
I have got today 512 when using sensitive. Did you change it now (I had downloaded new version yestarday). On the other hand I can edit OverlapMhap.pm. Thanks.
I remember from PBcR that I had to increase it to 1024 as at 512 it was giving bad results. When using Mhap 1.5 in Canu with default 756 results were great. Over all Canu looks to work better than PBcR thank you very much !:-)
